### PR TITLE
RESTWS-716: Orders resource returns no results when only status is specified

### PR DESCRIPTION
**Pull request by IamMujuziMoses**
## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-716

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java [hunk 1]
```diff
@@ -273,9 +274,8 @@ protected PageableResult doSearch(RequestContext context) throws ResponseExcepti
 			else {
 				return new NeedsPaging<Order>(orders, context);
 			}
-		}
-		
-		return new EmptySearchResult();
+		} else throw new InvalidSearchException("Please specify patient parameter with valid patientUuid");
```
Comment (dkayiwa) [inline omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java:277]: Can you format this like the rest of the code base?

Comment (IamMujuziMoses) [reply omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java:277]: Yes, sure

**Comment by dkayiwa** [comment]
@IamMujuziMoses can you add a test in the OrderController1_10Test?

**Comment by IamMujuziMoses** [comment]
@dkayiwa, I have added a test in OrderController1_10Test, is that the test you required, or something else?!

**Comment by dkayiwa** [comment]
@IamMujuziMoses the test should deal with what you are fixing. In other words, a test which would fail if your fix is not in.

File: omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java [hunk 2]
```diff
@@ -589,4 +590,38 @@ public void invalidCareCenterShouldThrowException() throws Exception {
...
+		int activeTestOrderCount = activeTestOrders.size();
+
+		// order service should return all active orders with patientUuid specified
+		MockHttpServletRequest req = newGetRequest(getURI(),
+				new Parameter("patient", PATIENT_UUID),
+				new Parameter("status", "active")
+		);
+		SimpleObject orders = deserialize(handle(req));
+		ArrayList<Object> resp = (ArrayList<Object>) PropertyUtils.getProperty(orders, "results");
+		assertEquals(activeTestOrderCount + activeDrugOrderCount, resp.size());
```
Comment (ManojLL) [inline omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java:624]: @IamMujuziMoses we can use `activeTestOrders.size() + activeTestOrders.size()` without creating separate variables.

Comment (ManojLL) [inline omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java:624]: @IamMujuziMoses we can use ` activeDrugOrders.size() + activeTestOrders.size()` without creating two variables. 
```suggestion
		assertEquals( activeDrugOrders.size() + activeTestOrders.size(), resp.size());
```

File: omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java [hunk 3]
```diff
@@ -273,9 +274,10 @@ protected PageableResult doSearch(RequestContext context) throws ResponseExcepti
 			else {
 				return new NeedsPaging<Order>(orders, context);
 			}
+		} else {
+			throw new InvalidSearchException("Please specify patient parameter with valid patientUuid");
```
Comment (dkayiwa) [inline omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java:278]: The error message reads as if one specified a patient parameter but with an invalid uuid.

File: omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java [hunk 4]
```diff
@@ -589,4 +590,36 @@ public void invalidCareCenterShouldThrowException() throws Exception {
 		        );
 		handle(req);
 	}
+
+	@Test(expected = InvalidSearchException.class)
+	public void doSearch_shouldReturnExceptionNotNullSetIfNoPatientUuidIsSpecified() throws Exception {
```
Comment (dkayiwa) [inline omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java:595]: I do not seem to understand the meaning of this method name.

File: omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java [hunk 5]
```diff
@@ -589,4 +590,36 @@ public void invalidCareCenterShouldThrowException() throws Exception {
...
+	public void shouldGetOrdersByPatientUuid() throws Exception {
+
+		Patient patient = patientService.getPatientByUuid(PATIENT_UUID);
+
+		// get all drug orders
+		List<Order> activeDrugOrders = orderService.getActiveOrders(patient, orderService.getOrderTypeByName("Drug order"),
+				null, null);
+
+		// get all test orders
+		List<Order> activeTestOrders = orderService.getActiveOrders(patient, orderService.getOrderTypeByName("Test order"),
```
Comment (dkayiwa) [inline omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java:612]: How many active test orders do you expect?

File: omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java [hunk 6]
```diff
@@ -589,4 +590,36 @@ public void invalidCareCenterShouldThrowException() throws Exception {
...
+	@Test(expected = InvalidSearchException.class)
+	public void doSearch_shouldReturnExceptionNotNullSetIfNoPatientUuidIsSpecified() throws Exception {
+		MockHttpServletRequest req = newGetRequest(getURI(),
+				new Parameter("status", "active")
+		);
+		handle(req);
+	}
+
+	@Test
+	public void shouldGetOrdersByPatientUuid() throws Exception {
```
Comment (dkayiwa) [inline omod-1.10/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_10/OrderController1_10Test.java:603]: Do you mind explaining how this test is related to the ticket?

**Pull request merged**